src/View/ViewBuilder.php

```diff
@@ -0,0 +1,332 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.1.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\View;
+
+use Cake\Core\App;
+use Cake\Event\EventManager;
+use Cake\Network\Request;
+use Cake\Network\Response;
+use Cake\View\View;
+use Cake\View\Exception\MissingViewException;
+
+
+/**
+ * Provides an API for iteratively building a view up.
+ *
+ * Once you have configured the view and established all the context
+ * you can create a view instance with `build()`.
+ */
+class ViewBuilder
+{
+    /**
+     * The subdirectory to the view.
+     *
+     * @var string
+     */
+    protected $viewPath;
+
+    /**
+     * The template file to render.
+     *
+     * @var string
+     */
+    protected $template;
+
+    /**
+     * The plugin name to use.
+     *
+     * @var string
+     */
+    protected $plugin;
+
+    /**
+     * The theme name to use.
+     *
+     * @var string
+     */
+    protected $theme;
+
+    /**
+     * The layout name to render.
+     *
+     * @var string
+     */
+    protected $layout;
+
+    /**
+     * Whether or not autoLayout should be enabled.
+     *
+     * @var bool
+     */
+    protected $autoLayout;
+
+    /**
+     * The layout path to build the view with.
+     *
+     * @var string
+     */
+    protected $layoutPath;
+
+    /**
+     * The view variables to use
+     *
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * The view variables to use
+     *
+     * @var string
+     */
+    protected $className;
+
+    /**
+     * The view variables to use
+     *
+     * @var array
+     */
+    protected $options = [];
+
+    /**
+     * The helpers to use
+     *
+     * @var array
+     */
+    protected $helpers = [];
+
+    /**
+     * Get/set path for view files.
+     *
+     * @param string|null $path Path for view files. If null returns current path.
+     * @return string|$this
+     */
+    public function viewPath($path = null)
```

Comment:
Done. I didn't rename the view property/method, but the new methods have better names.